internal/nexodus/nexodus.go

```diff
@@ -335,8 +340,25 @@ func (ax *Nexodus) Start(ctx context.Context, wg *sync.WaitGroup) error {
 		util.RunPeriodically(ctx, pollInterval, func() {
 			if err := ax.Reconcile(ax.organization, false); err != nil {
 				// TODO: Add smarter reconciliation logic
-				// to handle disconnects and/or timeouts etc...
 				ax.logger.Errorf("Failed to reconcile state with the nexodus API server: %v", err)
+				// if the token grant becomes invalid expires refresh or exit depending on the onboard method
+				if strings.Contains(err.Error(), invalidTokenGrant.Error()) {
+					if ax.username != "" {
+						c, err := client.NewClient(ctx, ax.controllerURL.String(), func(msg string) {
+							ax.SetStatus(NexdStatusAuth, msg)
+						}, option)
+						if err != nil {
+							ax.logger.Errorf("Failed to reconnect to the api-server, retrying in %v seconds: %v", pollInterval, err)
```

Comment:
Just a suggestion : I think it's good to probably slow down the retry attempt with every failed attempt

Reply:
Agree! I was thinking the whole `util.Waitgroup` should have a backoff though, that captures the other reconcile group for the discovery node as well to relieve unnecessary pressure on the controller. This retry will only be hit like once a month as far as I can tell with the Keycloack defaults. We need to capture other errors tbd as well.

Reply:
I think once in month probe doesn't need special treatment as of now. Sounds good to me.